practice-tool/src/widgets/target.rs

```diff
@@ -289,6 +317,40 @@ impl Widget for Target {
         pbar("Frost", frost, frost_max, COLOR_FROST);
         pbar("Sleep", sleep, sleep_max, COLOR_SLEEP);
         pbar("Mad", mad, mad_max, COLOR_MAD);
+
+        if let Some([x, y, z, r1, _r2]) = player_chunk_position {
+            let distance =
+                ((position.x - x).powf(2.) + (position.y - y).powf(2.) + (position.z - z).powf(2.))
+                    .sqrt();
+
+            let angle = f32::atan((x - position.x) / (z - position.z));
+            let my_angle = f32::asin(r1) * 2.0;
+            let enemy_angle = f32::asin(position.angle1) * 2.0;
+
+            let my_angle_display = my_angle * 180.0 / PI;
+            let enemy_angle_display = enemy_angle * 180.0 / PI;
+
+            let mut relative_angle = if (z - position.z) >= 0.0 {
+                (enemy_angle - angle) * 180.0 / PI
+            } else {
+                (enemy_angle - angle) * 180.0 / PI + 180.0
+            };
+
+            if relative_angle < 0.0 {
+                relative_angle = relative_angle + 360.0;
+            }
+
+            if relative_angle > 360.0 {
+                relative_angle = relative_angle - 360.0;
+            }
+
+            relative_angle = relative_angle - 180.0;
+
+            ui.text(format!("{distance:>6.3}m {relative_angle:>6.4}deg",));
+            ui.text(
+                format!("{my_angle_display:>6.4}myangle {enemy_angle_display:>6.4}enemyangle",),
+            );
```

Comment:
```suggestion
            ui.text(format!("{distance:>6.3}m {my_angle_display:>6.3}deg",));
```

Let's keep it like this for now, meanwhile I'm going to cook an angle visualization before merging this into a new release.